aas-gui/Frontend/aas-web-gui/src/components/SubmodelPlugins/ProductCarbonFootprint.vue

```diff
@@ -0,0 +1,161 @@
+            const total = pcfValue + tcfValue;
+            return total ? (pcfValue / total) * 100 : 0;
+        },
+        tcfPercentage() {
+            const pcfValue = parseFloat(this.productIsoValue) || 0;
+            const tcfValue = parseFloat(this.transportENValue) || 0;
+            const total = pcfValue + tcfValue;
+            return total ? (tcfValue / total) * 100 : 0;
+        },
+        pcfValueColor() {
+            const isDark = this.$vuetify.theme.global.current.dark; // Check if the current theme is dark
+            const themeColors = isDark ? this.$vuetify.theme.themes.dark.colors : this.$vuetify.theme.themes.light.colors;
+            return themeColors.pcf; // Return the pcf color or a default color if not found
+        },
+        tcfValueColor() {
+            const isDark = this.$vuetify.theme.global.current.dark; // Check if the current theme is dark
+            const themeColors = isDark ? this.$vuetify.theme.themes.dark.colors : this.$vuetify.theme.themes.light.colors;
+            return themeColors.tcf; // Return the pcf color or a default color if not found
+        },
+        fingersColor() {
+            const isDark = this.$vuetify.theme.global.current.dark; // Check if the current theme is dark
+            const themeColors = isDark ? this.$vuetify.theme.themes.dark.colors : this.$vuetify.theme.themes.light.colors;
+            return themeColors.fingers; // Return the fingers color or a default color if not found
+        }
+    },
+    methods: {
+        // Function to initialize the Product Carbon Footprint Plugin
+        initializeProductCarbonFootprint() {
+            // Check if a Node is selected
+            if (Object.keys(this.submodelElementData).length == 0) {
+                this.productCarbonFootprintData = {}; // Reset the productCarbonFootprint Data when no Node is selected
+                return;
+            }
+            let productCarbonFootprintData = { ...this.submodelElementData }; // create local copy of the productCarbonFootprintData
+            this.productCarbonFootprintData = this.calculateSubmodelElementPathes(productCarbonFootprintData, this.SelectedNode.path); // Set the DigitalNameplate Data
+            this.productIsoValue = this.submodelElementData?.submodelElements?.[0]?.value?.[1]?.value;
+        },
+        intializeTransportCarbonFootprint() {
+            if (Object.keys(this.submodelElementData).length == 0) {
+                this.transportCarbonFootprintData = {}; // Reset the productCarbonFootprint Data when no Node is selected
+                return;
+            }
+            let transportCarbonFootprintData = { ...this.submodelElementData };
+            this.transportCarbonFootprintData = this.calculateSubmodelElementPathes(transportCarbonFootprintData, this.SelectedNode.path);
+            this.transportENValue = this.submodelElementData?.submodelElements?.[1]?.value?.[1]?.value;
```

Comment:
This assumes the TCF value is in the same place all the time and only one TCF is supported. In reality, the cardinality is 0..*